presto-main/src/test/java/io/prestosql/operator/scalar/AbstractTestRegexpFunctions.java

```diff
@@ -236,4 +237,50 @@ public void testRegexpSplit()
         assertFunction("REGEXP_SPLIT(',,a,,,b,c,d,,', ',')", new ArrayType(createVarcharType(13)), ImmutableList.of("", "", "a", "", "", "b", "c", "d", "", ""));
         assertFunction("REGEXP_SPLIT(',,,', ',')", new ArrayType(createVarcharType(3)), ImmutableList.of("", "", "", ""));
     }
+
+    @Test
+    public void testRegexpCount()
+    {
+        assertFunction("REGEXP_COUNT('a.b:c;d', '[\\.:;]')", IntegerType.INTEGER, 3);
+        assertFunction("REGEXP_COUNT('a.b:c;d', '\\.')", IntegerType.INTEGER, 1);
+        assertFunction("REGEXP_COUNT('a.b:c;d', ':')", IntegerType.INTEGER, 1);
+        assertFunction("REGEXP_COUNT('a,b,c', ',')", IntegerType.INTEGER, 2);
+        assertFunction("REGEXP_COUNT('a1b2c3d', '\\d')", IntegerType.INTEGER, 3);
+        assertFunction("REGEXP_COUNT('a1b2346c3d', '\\d+')", IntegerType.INTEGER, 3);
+        assertFunction("REGEXP_COUNT('abcd', 'x')", IntegerType.INTEGER, 0);
+        assertFunction("REGEXP_COUNT('abcd', '')", IntegerType.INTEGER, 0);
+        assertFunction("REGEXP_COUNT('', 'x')", IntegerType.INTEGER, 0);
+    }
+
+    @Test
+    public void testRegexpPosition()
```

Comment:
add tests with non-unicode positions. I think re2J might actually count bytes and not code-points.

Reply:
> add tests with non-unicode positions. I think re2J might actually count bytes and not code-points.

I added some tests for non-english string. I get the code-point like this:
```
source.slice(0, matcher.start() + startBytePosition).toStringUtf8().length() + 1
```
It's not Efficiently,  any suggestion?

Reply:
I'm not sure it can be improved without explicit support in `RE2J` library.

Why do you need `+1` here?

Reply:
"source.slice(0, matcher.start() + startBytePosition)" returns the string before the position mathed because mather.getBegin() returns position starting from zero.